diff --git a/socfpga_cyclone5_de0_nano_soc.dts.orig b/socfpga_cyclone5_de0_nano_soc.dts
index c8f051fb2b..264f0f7361 100644
--- a/arch/arm/boot/dts/socfpga_cyclone5_de0_nano_soc.dts
+++ b/arch/arm/boot/dts/socfpga_cyclone5_de0_nano_soc.dts
@@ -32,13 +32,23 @@
 	};
 
 	leds {
-		compatible = "gpio-leds";
+		compatible = "my-gpio-led";
 		hps0 {
 			label = "hps_led0";
 			gpios = <&portb 24 0>;
 			linux,default-trigger = "heartbeat";
 		};
 	};
+
+	buttons {
+		compatible = "my-gpio-button";
+		btn0 {
+			label = "hps_btn0";
+			gpios = <&portb 25 0>;
+			debounce-interval = <50>;
+			wakeup-source;
+		};
+	};
 };
 
 &gmac1 {
